derp/derp_client.go

```diff
@@ -201,6 +202,40 @@ func (c *Client) send(dstKey key.Public, pkt []byte) (ret error) {
 	return c.bw.Flush()
 }
 
+func (c *Client) ForwardPacket(srcKey, dstKey key.Public, pkt []byte) (err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("derp.ForwardPacket: %w", err)
+		}
+	}()
+
+	if len(pkt) > MaxPacketSize {
+		return fmt.Errorf("packet too big: %d", len(pkt))
+	}
+
+	c.wmu.Lock()
+	defer c.wmu.Unlock()
+
+	timer := time.AfterFunc(5*time.Second, c.writeTimeoutFired)
```

Comment:
Why not setting a deadline on the conn?

Reply:
I thought this would be more efficient and wouldn't clobber any existing write deadline. *shrug*

Mostly the clobbering.

Really we should have contexts plumbed. There's a TODO for that elsewhere in here.